lib/spack/spack/test/zz/bindist_rel.py

```diff
@@ -0,0 +1,379 @@
+def install_dir_non_default_layout(tmpdir):
+    """Hooks a fake install directory with a non-default directory layout"""
+    real_store = spack.store.store
+    real_layout = spack.store.layout
+    spack.store.store = spack.store.Store(str(tmpdir.join('opt')))
+    spack.store.layout = YamlDirectoryLayout(str(tmpdir.join('opt')),
+                                             path_scheme=ndef_install_path_scheme)  # noqa: E501
+    yield spack.store
+    tmpdir.join('opt').remove()
+    spack.store.store = real_store
+    spack.store.layout = real_layout
+
+
+@pytest.mark.maybeslow
+@pytest.mark.usefixtures('gpg_directory', 'default_config',
+                         'install_dir_default_layout', 'cache_directory')
+def test_buildcache_create_install_rel(tmpdir, mirror_directory_rel,
+                                       mock_stage):
+    """
+    Test the creation and installation of buildcaches with relative
+    rpaths into the default directory layout scheme.
+    """
+    global mirror_path_rel
+    mirror_path_rel = mirror_directory_rel
+    mparser = argparse.ArgumentParser()
+    mirror.setup_parser(mparser)
+    margs = mparser.parse_args(
+        ['add', '--scope', 'site', 'test-mirror-rel', 'file://%s' % mirror_path_rel])
+    mirror.mirror(mparser, margs)
+    margs = mparser.parse_args(['list'])
+    mirror.mirror(mparser, margs)
+    cparser = argparse.ArgumentParser()
+    compiler.setup_parser(cparser)
+    cargs = cparser.parse_args(['add', '--scope', 'site'])
+    compiler.compiler(cparser, cargs)
+    cargs = cparser.parse_args(['list', '--scope', 'site'])
+    compiler.compiler(cparser, cargs)
+    cargs = cparser.parse_args(['info', 'clang'])
+    compiler.compiler(cparser, cargs)
+    cargs = cparser.parse_args(['info', 'gcc'])
+    compiler.compiler(cparser, cargs)
+
+    rparser = argparse.ArgumentParser()
+    repocmd.setup_parser(rparser)
+    rargs = rparser.parse_args(['add', spack.paths.packages_path])
+    repocmd.repo(rparser, rargs)
+    njobs = spack.config.get('config:build_jobs')
+    if not njobs:
+        spack.config.set('config:build_jobs', 4, scope='user')
+    extensions = spack.config.get('config:template_dirs')
+    if not extensions:
+        spack.config.set('config:template_dirs',
+                         [os.path.join(spack.paths.share_path, 'templates')],
+                         scope='user')
+    espec = Spec('environment-modules')
+    espec.concretize()
+    zspec = Spec('zlib')
+    zspec.concretize()
+    pspec = Spec('patchelf')
+    pspec.concretize()
+
+    margs = mparser.parse_args(['list'])
+    mirror.mirror(mparser, margs)
+
+    # setup argument parser
+    parser = argparse.ArgumentParser()
+    buildcache.setup_parser(parser)
+
+    # set default builcache args
+    create_args = ['create', '-a', '-r', '-d', mirror_path_rel, espec.name]
+    install_args = ['install', '-a', espec.name]
+
+    # Create a private key to sign package with if gpg2 available
+    if has_gnupg2():
```

Comment:
I think testing of gpg signing can be moved into a separate test: i.e. `bindist_def` and `bindist_rel` both do these checks in their tests. I think in general they can proceed with creating unsigned build caches (since that is faster), and a single test could be used to verify the gpg-signing part (if that hasn't been done already).

Reply:
I initially had both in one file but made them separate because the download cache directory for buildcaches was not being cleared out between default and relative tests. This did not test the relative functionality because it was using the old default buildcaches.

Reply:
I'm confused as this comment (https://github.com/spack/spack/pull/13797#discussion_r365400287) doesn't specifically seem to address the comment it is attached to (https://github.com/spack/spack/pull/13797#discussion_r365028818). Here I am saying you can simplify the tests (and perhaps make them run faster) by removing all GPG signing and generating unsigned caches for these new tests (while we do want to test GPG signing, I don't think that has to be done here).

Reply:
My main objective was to increase coverage.

OK I will need to add flags to skip signature checking on linux where gpg2 is usually available. I will also have to skip `spack install` which requires a signed buildcache in order to install.